tests/contract/knative-channel-kafka/kafka_channel_test.go

```diff
@@ -47,24 +59,99 @@ func TestKnativeEventingKafkaChannelAcceptance(t *testing.T) {
 	// delete the Kafka channel if existed before to make sure that
 	// the new channel to be created has the correct structure and data
 	if err := deleteChannelIfExistsAndWaitUntilDeleted(t, interrupted, kafkaClient, name, 5*time.Second, 10, retry.FixedDelay); err != nil {
-		t.Fatalf("test failed with error: %s", err)
+		log.Printf("test failed with error: %s", err)
+		t.FailNow()
 	}
 
 	// create a Kafka channel
 	if _, err := kafkaClient.Create(newKafkaChannel(name, namespace)); err != nil {
-		t.Fatalf("cannot create a Kafka channel: %s: error: %v", name, err)
+		log.Printf("cannot create a Kafka channel: %s: error: %v", name, err)
+		t.FailNow()
 	} else {
-		t.Logf("created Kafka channel: %s", name)
+		log.Printf("created Kafka channel: %s", name)
 	}
 
 	// assert the Kafka channel status to be ready
-	if err := checkChannelReadyWithRetry(t, interrupted, kafkaClient, name, 5*time.Second, 10, retry.FixedDelay); err != nil {
-		t.Fatalf("test failed with error: %s", err)
-	} else {
-		t.Logf("test finished successfully")
+	readyKafkaChannel, err := checkChannelReadyWithRetry(t, interrupted, kafkaClient, name, 5*time.Second, 10, retry.FixedDelay)
+	if err != nil {
+		log.Printf("test failed with error: %s", err)
+		t.FailNow()
 	}
 
-	// TODO(marcobebway) extend the test to assert event delivery also works using the Kafka channel https://github.com/kyma-project/kyma/issues/7015.
+	// send CE to Kafka channel
+	target := readyKafkaChannel.Status.Address.URL
+	ceClient := createCloudEventsClient(t, target)
+	event := createCloudEvent(t)
+	sendEventUntilReceived(t, interrupted, event, target, ceClient)
+
+	log.Printf("test finished successfully")
+}
+
+// sendEventUntilReceived sends a given CE to the target and retry until the event was successfully received (2xx status code).
+func sendEventUntilReceived(t *testing.T, interrupted chan bool, event cloudevents.Event, target *apis.URL, ceClient client.Client) {
+	err := retry.Do(func() error {
+		select {
+		case <-interrupted:
+			log.Printf("cannot continue, test was interrupted")
+			return retry.Unrecoverable(fmt.Errorf("cannot continue, test was interrupted"))
+		default:
+			// send an CE event to Kafka channel
+			log.Printf("sending cloudevent to Kafka channel: %q", target)
+			rctx, _, err := ceClient.Send(context.Background(), event)
+			if err != nil {
+				return err
+			}
+			rtctx := cloudevents.HTTPTransportContextFrom(rctx)
+			log.Printf("received status code: %d", rtctx.StatusCode)
+			if !is2XXStatusCode(rtctx.StatusCode) {
+				return fmt.Errorf("received non 2xx status code: %d", rtctx.StatusCode)
+			}
+			return nil
+		}
+	}, retry.Attempts(24), retry.Delay(time.Second*5), // 120=24*5 seconds
```

Comment:
fixeddelay will help: otherwise the delay is 2^(attempt)*delay
e.g. 2^24*5 = 83*10^6 s 
imho thats too much